sourcegraph/sourcegraph.proto

```diff
@@ -54,13 +54,17 @@ message ListOptions {
 	int32 page = 2 [(gogoproto.moretags) = "url:\",omitempty\""];
 }
 
-// ListResponse specifies general pagination response when fetching a list of results.
+// ListResponse specifies a general paginated response when fetching a list of results.
 message ListResponse {
-	// HasMore is true if there are more entries available after the returned page.
-	bool has_more = 1 [(gogoproto.moretags) = "url:\",omitempty\""];
-
 	// Total is the total number of results in the list.
-	int32 total = 2 [(gogoproto.moretags) = "url:\",omitempty\""];
+	int32 total = 1 [(gogoproto.moretags) = "url:\",omitempty\""];
+}
+
+// StreamResponse specifies a paginated response where the total number of results
+// that can be returned is unbounded or unknown.
```

Comment:
Do you think it's worth mentioning, "or too expensive to compute?" I feel that's a pretty common case, at least right now. The number of results is neither unbounded nor unknown, it's just hard to compute and not worth it.

However, this would also be great to use for unbounded/unknown cases.